ansible/pbTestScripts/buildJDK.sh

```diff
@@ -1,26 +1,118 @@
+			"--openj9" | "-j9" )
+				VARIANT=openj9;;
+			"--clean-workspace" | "-c" )
+				CLEAN_WORKSPACE=true;;
+			"--help" | "-h" )
+				usage; exit 0;;
+			*) echo >&2 "Invalid option: ${opt}"; echo "This option was unrecognised."; usage; exit 1;;
+		esac
+	done
 
-export PATH=/usr/local/bin/:$PATH
-if [ -z "${WORKSPACE:-}" ]; then
-	echo "WORKSPACE not found, setting it as environment variable 'HOME'"
-	WORKSPACE=$HOME
-fi
+	checkJDKVersion $JAVA_TO_BUILD
+	if [ -z "${WORKSPACE:-}" ]; then
+        	echo "WORKSPACE not found, setting it as environment variable 'HOME'"
+        	WORKSPACE=$HOME
+	fi
+	if [ $CLEAN_WORKSPACE == true ]; then
+		echo "Cleaning workspace"
+		rm -rf $WORKSPACE/openjdk-build
+	fi
+}
+
+usage() {
+	echo "Usage: ./buildJDK.sh <options>
+	
+	Options:
+		--version | -v		Specify the JDK version to build
+		--URL | -u		Specify the github URL to clone openjdk-build from
+		--openj9 | -j9		Builds openJ9, instead of hotspot
+		--clean-workspace | -c 	Removes old openjdk-build folder before cloning
+		--help | -h		Shows this message
+		
+	If not specified, JDK8-HS will be built with the standard openjdk-build repo"
+	echo
+	showJDKVersions
+}
+
+showJDKVersions() {
+	echo "Currently supported JDK versions: 
+		- JDK8
+		- JDK9
+		- JDK10
+		- JDK11
+		- JDK12
+		- JDK13
+		- JDK14"
+	echo
+}
 
+checkJDKVersion() {
+        local jdk=$1
+        case "$jdk" in
+                "jdk8u" | "jdk8" | "8" | "8u" )
+                        JAVA_TO_BUILD="jdk8u";;
+                "jdk9u" | "jdk9" | "9" | "9u" )
+                        JAVA_TO_BUILD="jdk9u";;
+                "jdk10u" | "jdk10" | "10" | "10u" )
+                        JAVA_TO_BUILD="jdk10u";;
+                "jdk11u" | "jdk11" | "11" | "11u" )
+                        JAVA_TO_BUILD="jdk11u";;
+                "jdk12u" | "jdk12" | "12" | "12u" )
+                        JAVA_TO_BUILD="jdk12u";;
+                "jdk13u" | "jdk13" | "13" | "13u" )
+                        JAVA_TO_BUILD="jdk13u";;
+                "jdk14u" | "jdk14" | "14" | "14u" )
+                        JAVA_TO_BUILD="jdk14u";;
+                *)
+                        echo "Not a valid JDK Version" ; showJDKVersions; exit 1;;
+        esac
+}
+
+cloneRepo() {
+	local branch=""
+	IFS='/' read -r -a urlArray <<< "$GIT_URL"
+	if [ -d $WORKSPACE/openjdk-build ]; then
+		echo "Found existing openjdk-build folder"
+		cd $WORKSPACE/openjdk-build && git pull
+	elif [ ${urlArray[@]: -2:1} == 'tree' ]; then
+		GIT_URL=""
+		echo "Branch detected"
+		branch=${urlArray[@]: -1:1}
+		unset 'urlArray[${#urlArray[@]}-1]'
+		unset 'urlArray[${#urlArray[@]}-1]'
+		for urlPart in "${urlArray[@]}"
+		do
+			GIT_URL="$GIT_URL$urlPart/"
+		done
+		git clone -b $branch $GIT_URL $WORKSPACE/openjdk-build
+	else
+		echo "No branch detected"
+		git clone $GIT_URL $WORKSPACE/openjdk-build
+	fi
+}
+
+# Default values
+export JAVA_TO_BUILD=jdk8u
+export PATH=/usr/local/bin/:$PATH
 export TARGET_OS=linux
-export VARIANT=openj9
```

Comment:
We should leave this as openj9 as it has the higher level of prereqs so should be the one to use unless there's a good reason not to